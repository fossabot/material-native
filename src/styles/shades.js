'use strict';
/**
 * Translucent black and white colors used for text, icons, and dividers.
 * https://material.io/guidelines/style/color.html#color-usability
 */

const dark = (opacity) => `rgba(0, 0, 0, ${opacity})`;
const light = (opacity) => `rgba(255, 255, 255, ${opacity})`;

function makeShades(genColor, opacities) {
	const shades = {};

	for ( const shade in opacities ) {
		shades[shade] = genColor(opacities[shade]);
	}

	return shades;
}

export const opacity = {
	// Translucent blacks
	dark: {
		primaryText: 0.87,
		secondaryText: 0.54,
		disabledText: 0.38,
		hintText: 0.38,
		dividers: 0.12,
		activeIcon: 0.54,
		inactiveIcon: 0.38,
	},
	// Translucent whites
	light: {
		primaryText: 1,
		secondaryText: 0.70,
		disabledText: 0.50,
		hintText: 0.50,
		dividers: 0.12,
		activeIcon: 1,
		inactiveIcon: 0.5,
	},
};

export default ({
	opacity,
	dark: makeShades(dark, opacity.dark),
	light: makeShades(light, opacity.light),
});
