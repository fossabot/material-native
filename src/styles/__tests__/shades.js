'use strict';
import shades from '../shades';

test('shades module matches snapshot', () => {
	expect(shades).toMatchSnapshot();
});
