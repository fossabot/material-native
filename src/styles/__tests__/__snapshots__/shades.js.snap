// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shades module matches snapshot 1`] = `
Object {
  "dark": Object {
    "activeIcon": "rgba(0, 0, 0, 0.54)",
    "disabledText": "rgba(0, 0, 0, 0.38)",
    "dividers": "rgba(0, 0, 0, 0.12)",
    "fieldBottomLine": "rgba(0, 0, 0, 0.42)",
    "hintText": "rgba(0, 0, 0, 0.38)",
    "inactiveIcon": "rgba(0, 0, 0, 0.38)",
    "primaryText": "rgba(0, 0, 0, 0.87)",
    "secondaryText": "rgba(0, 0, 0, 0.54)",
    "statusBar": "rgba(0, 0, 0, 0.2)",
  },
  "light": Object {
    "activeIcon": "rgba(255, 255, 255, 1)",
    "disabledText": "rgba(255, 255, 255, 0.5)",
    "dividers": "rgba(255, 255, 255, 0.12)",
    "fieldBottomLine": "rgba(255, 255, 255, 0.7)",
    "hintText": "rgba(255, 255, 255, 0.5)",
    "inactiveIcon": "rgba(255, 255, 255, 0.5)",
    "primaryText": "rgba(255, 255, 255, 1)",
    "secondaryText": "rgba(255, 255, 255, 0.7)",
    "statusBar": "rgba(255, 255, 255, 0.7)",
  },
  "opacity": Object {
    "dark": Object {
      "activeIcon": 0.54,
      "disabledText": 0.38,
      "dividers": 0.12,
      "fieldBottomLine": 0.42,
      "hintText": 0.38,
      "inactiveIcon": 0.38,
      "primaryText": 0.87,
      "secondaryText": 0.54,
      "statusBar": 0.2,
    },
    "light": Object {
      "activeIcon": 1,
      "disabledText": 0.5,
      "dividers": 0.12,
      "fieldBottomLine": 0.7,
      "hintText": 0.5,
      "inactiveIcon": 0.5,
      "primaryText": 1,
      "secondaryText": 0.7,
      "statusBar": 0.7,
    },
  },
}
`;
